src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,924 @@
+
+Data is made available on the backend USGS file explorer. For example,
+dekadal temporally smooth NDVI data for West Africa is available at
+`this link
+<https://edcintl.cr.usgs.gov/downloads/sciweb1/shared/fews/web/africa/west/dekadal/emodis/ndvi_c6/temporallysmoothedndvi/downloads/monthly/>`_
+
+The products include temporally smoothed NDVI, median anomaly,
+difference from the previous year, and median anomaly
+presented as a percentile.
+
+Data by USGS is published quickly after the dekad.
+After about 1 month this data is updated with temporal smoothing
+and error correction for cloud cover.
+"""
+
+# TODO: add progress bar
+import logging
+from datetime import date
+from io import BytesIO
+from pathlib import Path
+from typing import List, Tuple, Union, cast
+from urllib.error import HTTPError
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import geopandas as gpd
+import pandas as pd
+import regex as re
+import rioxarray  # noqa: F401
+import xarray as xr
+from typing_extensions import Literal
+
+import aatoolbox.utils.raster  # noqa: F401
+from aatoolbox.config.countryconfig import CountryConfig
+from aatoolbox.datasources.datasource import DataSource
+from aatoolbox.utils._dates import (
+    _compare_dekads_gt,
+    _compare_dekads_lt,
+    _date_to_dekad,
+    _dekad_to_date,
+    _expand_dekads,
+)
+
+# TODO: use check_file_existence once the fix is merged
+# that allows it to work on class methods (PR is open)
+# from aatoolbox.utils.io import check_file_existence
+
+logger = logging.getLogger(__name__)
+
+_VALID_AREAS = Literal[
+    "north_africa",
+    "east_africa",
+    "southern_africa",
+    "west_africa",
+    "central_asia",
+    "yemen",
+    "central_america",
+    "hispaniola",
+]
+_VALID_TYPES = Literal[
+    "smoothed", "percent_median", "median_anomaly", "difference"
+]
+
+
+class _UsgsNdvi(DataSource):
+    """Base class to retrieve USGS NDVI data.
+
+    Parameters
+    ----------
+    country_config : CountryConfig
+        Country configuration
+    coverage_area : str
+        Area of coverage
+    data_type : str
+        Data type
+    start_date : Union[date, str, Tuple[int], None]
+        Start date. Can be passed as a ``datetime.date``
+        object and the relevant dekad will be determined,
+        as a date string in ISO8601 format, or as a
+        year-dekad tuple, i.e. (2020, 1).
+    end_date : Union[date, str, Tuple[int], None]
+        End date. Can be passed as a ``datetime.date``
+        object and the relevant dekad will be determined,
+        as a date string in ISO8601 format, or as a
+        year-dekad tuple, i.e. (2020, 1).
+    processed_file_suffix : str
+        Suffix for processed file.
+    """
+
+    def __init__(
+        self,
+        country_config: CountryConfig,
+        coverage_area: _VALID_AREAS,
+        data_type: _VALID_TYPES,
+        start_date: Union[date, str, Tuple[int], None] = None,
```

Comment:
Fair question. However, since the tuple format is for dekads, I thought providing the option to pass in more user friendly dates would be easier. For instance, If you want all data from July 1 onward, I would find it much easier to pass `2020-07-01` as a string than figuring out what the year and dekad is. I've actually gotten all the `mypy` stuff running with checks, so let me know what you think.

Reply:
Yeah I would agree that that is easier while we prob still want the option of the dekad number as well. So would leave agree to leave as it is.